Strip whitespace from markers in INI config

Perlence opened the pull request:
Attempts to resolve #2856.

nicoddemus commented on CHANGELOG.rst:
```diff
@@ -25,6 +25,9 @@ Bug Fixes
   _fixtureinfo attribute (e.g. doctests) (`#2788
   <https://github.com/pytest-dev/pytest/issues/2788>`_)
 
+- Strip whitespace from the markers in INI config (`#2856
```
We use `towncrier` for changelog management to avoid conflicts, so instead of this please create a file `changelog/2856.bugfix` with this text:

```
Strip whitespace from marker names when reading them from INI config.
```

Perlence replied:
Oh, sorry! I should have paid more attention.

bilderbuchi commented:
For consistency, do you also have to update [this section](https://github.com/Perlence/pytest/blob/52aadcd7c126f05a7219db88ac8d3d233b3de5c3/_pytest/mark.py#L121-L122)? Although I haven't looked into why this one does not split on `(`, too.

Perlence commented:
@bilderbuchi I guess `(` is not stripped to show arguments of a mark in CLI.

pquentin commented:
For what it's worth, this change causes pytest to break in a non obvious way when there's a line without a colon, for example if the description of a marker was wrapped such as:

```
[pytest]
markers =
    a: A marker description that
       spans multiple lines
```

Perlence commented:
@pquentin I reverted the merge, but this doesn't work:

```python
def test_markers_option(testdir):
    testdir.makeini("""
        [pytest]
        markers =
            a1: this is a webtest marker
            a1some: another marker
            a1multi: another marker
                     with stuff
    """)
    result = testdir.runpytest("--markers", )
    result.stdout.fnmatch_lines([
        "*a1*this is a webtest*",
        "*a1some*another marker",
        "*a1multi*another marker*with stuff",
    ])
```

```
$ pytest -vvx -k markers_option
...
/home/sviatoslav/workspace/github.com/pytest/testing/test_mark.py:171: Failed
------------------------------------------------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------------------------------------------------
@pytest.mark.a1: this is a webtest marker

@pytest.mark.a1some: another marker

@pytest.mark.a1multi: another marker

------------------------------------------------------------------------------------------------------------------------------ Captured stderr call ------------------------------------------------------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/sviatoslav/workspace/github.com/pytest/_pytest/pytester.py", line 759, in runpytest_inprocess
    reprec = self.inline_run(*args, **kwargs)
  File "/home/sviatoslav/workspace/github.com/pytest/_pytest/pytester.py", line 732, in inline_run
    ret = pytest.main(list(args), plugins=plugins)
  File "/home/sviatoslav/workspace/github.com/pytest/_pytest/config.py", line 58, in main
    return config.hook.pytest_cmdline_main(config=config)
  File "/home/sviatoslav/workspace/github.com/pytest/_pytest/vendored_packages/pluggy.py", line 745, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/sviatoslav/workspace/github.com/pytest/_pytest/vendored_packages/pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/sviatoslav/workspace/github.com/pytest/_pytest/vendored_packages/pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "/home/sviatoslav/workspace/github.com/pytest/_pytest/vendored_packages/pluggy.py", line 614, in execute
    res = hook_impl.function(*args)
  File "/home/sviatoslav/workspace/github.com/pytest/_pytest/mark.py", line 122, in pytest_cmdline_main
    name, rest = line.split(":", 1)
ValueError: not enough values to unpack (expected 2, got 1)
```

nicoddemus commented:
@pquentin could you please open an issue describing the problem, including the error message you are getting? We can continue this discussion from there then. Thanks!
